core/src/main/java/io/ddf/etl/TransformationHandler.java

```diff
@@ -391,4 +390,36 @@ public static String RToSqlUdf(String RExp) {
   @Override public DDF oneHotEncoding(String inputColumn, String outputColumnName) throws DDFException {
     throw new UnsupportedOperationException();
   }
+
+  @Override
+  public DDF sort(List<String> columns, List<Boolean> ascending) throws DDFException {
+    if (columns == null || columns.isEmpty()) {
+      throw new DDFException("List of columns to sort is empty");
+    }
+    List<String> ddfColumns = this.getDDF().getSchema().getColumns().stream()
+            .map(Column::getName).collect(Collectors.toList());
+    for (String col : columns) {
+      if (!ddfColumns.contains(col)) {
+        throw new DDFException(String.format("Column to sort: %s is not in the DDF", col));
+      }
+    }
+
+    if (ascending == null) {
+      ascending = new ArrayList<>();
+    }
+
+    Iterator<String> columnIterator = columns.iterator();
+    Iterator<Boolean> ascendingIterator = ascending.iterator();
+    StringJoiner joiner = new StringJoiner(",", "", "");
+    while (columnIterator.hasNext()) {
+      boolean asc = ascendingIterator.hasNext() ? ascendingIterator.next() : true;
+      joiner.add(String.format("%s %s", columnIterator.next(), asc ? "asc" : "desc"));
+    }
+
+    String cmd = String.format("select * from %s order by ", this.getDDF().getSchema().getTableName()).concat(joiner.toString());
```

Comment:
better to use Spark DataFrame sort function rather than construct the sql which is error prone, see:
http://stackoverflow.com/questions/30332619/how-to-sort-by-column-in-descending-order-in-spark-sql